smart_open/s3.py

```diff
@@ -127,10 +131,22 @@ def open(
         )
     else:
         assert False, 'unexpected mode: %r' % mode
-
     return fileobj
 
 
+def _get(s3_object, version=None, **kwargs):
+    if version is not None:
+        kwargs['VersionId'] = version
+    try:
+        return s3_object.get(**kwargs)
+    except botocore.client.ClientError as error:
+        raise IOError(
```

Comment:
It's not common to have multiline exception messages in Python. Let's see if we can make this more brief.

What will the user see if we raise:

```python
raise IOError(str(error))
```

?

Reply:
Return exception
botocore.errorfactory.NoSuchKey: An error occurred (NoSuchKey) when calling the GetObject operation: The specified key does not exist.

Reply:
OK, let's do:

```python
raise IOError(
    'unable to access bucket: %r key: %r version: %r error: %s' % (
        s3_object.bucket_name, s3_object.key, version, error
    )
)
```